### PR TITLE
[Doc] Add cross library en doc.

### PR DESCRIPTION
**Pull request by RangiLyu**
Thanks for your contribution and we appreciate it a lot. The following instructions would make your pull request more healthy and more easily get feedback. If you do not understand some items, don't worry, just make the pull request and seek help from maintainers.

## Motivation

Please describe the motivation of this PR and the goal you want to achieve through this PR.

## Modification

Please briefly describe what modification is made in this PR.

## BC-breaking (Optional)

Does the modification introduce changes that break the backward-compatibility of the downstream repos?
If so, please describe how it breaks the compatibility and how the downstream projects should modify their code to keep compatibility with this PR.

## Use cases (Optional)

If this PR introduces a new feature, it is better to list some use cases here, and update the documentation.

## Checklist

1. Pre-commit or other linting tools are used to fix the potential lint issues.
2. The modification is covered by complete unit tests. If not, please add more unit test to ensure the correctness.
3. If the modification has potential influence on downstream projects, this PR should be tested with downstream projects, like MMDet or MMCls.
4. The documentation has been modified accordingly, like docstring or example tutorials.

### CODE REVIEW

File: docs/en/advanced_tutorials/cross_library.md [hunk 1]
```diff
@@ -1,3 +1,103 @@
 # Use modules from other libraries
 
-Coming soon. Please refer to [chinese documentation](https://mmengine.readthedocs.io/zh_CN/latest/examples/cross_library.html).
+Based on MMEngine's [Registry](registry.md) and [Config](config.md), users can build modules across libraries.
+For example, use [MMClassification](https://github.com/open-mmlab/mmclassification)'s backbones in [MMDetection](https://github.com/open-mmlab/mmdetection), or [MMDetection](https://github.com/open-mmlab/mmdetection)'s data transforms in [MMRotate](https://github.com/open-mmlab/mmrotate), or using [MMDetection](https://github.com/open-mmlab/mmdetection)'s detectors in [MMTracking](https://github.com/open-mmlab/mmtracking).
+
+Modules which registered in the same registry tree can be called across libraries by adding the **package name prefix** before the module's type in the config. Here are some common examples:
```
Comment (C1rN09) [inline docs/en/advanced_tutorials/cross_library.md:6]:
```suggestion
Modules registered in the same registry tree can be called across libraries by adding the **package name prefix** before the module's type in the config. Here are some common examples:
```

File: docs/en/advanced_tutorials/cross_library.md [hunk 2]
```diff
@@ -1,3 +1,103 @@
...
+    dict(type='ConvertBoxType', box_type_mapping=dict(gt_bboxes='rbox')),
+    dict(type='mmdet.Resize', scale=(1024, 2014), keep_ratio=True),
+    dict(type='mmdet.RandomFlip', prob=0.5),
+    dict(type='mmdet.PackDetInputs')
+]
+```
+
+## Use detector across libraries:
+
+Using an algorithm across libraries is a little bit complex.
```
Comment (C1rN09) [inline docs/en/advanced_tutorials/cross_library.md:60]:
```suggestion
Using a module from another library is a little bit complex.
```
Maybe 'module' here is more proper than 'algorithm'?

Comment (RangiLyu) [reply docs/en/advanced_tutorials/cross_library.md:60]: This example is about using a detection algorithm whose complexity differs from a single module.

Comment (C1rN09) [reply docs/en/advanced_tutorials/cross_library.md]: Makes sense to me

**Pull request merged**